lib/ansible/modules/cloud/amazon/ec2_instance.py

```diff
@@ -75,65 +85,123 @@
   network:
     description:
       - Either a dictionary containing the key 'interfaces' corresponding to a list of network interface IDs or
         containing specifications for a single network interface.
-      - If specifications for a single network are given, accepted keys are assign_public_ip (bool),
-        private_ip_address (str), ipv6_addresses (list), source_dest_check (bool), description (str),
-        delete_on_termination (bool), device_index (int), groups (list of security group IDs),
-        private_ip_addresses (list), subnet_id (str).
-      - I(network.interfaces) should be a list of ENI IDs (strings) or a list of objects containing the key I(id).
-      - Use the ec2_eni to create ENIs with special settings.
+      - Use the ec2_eni module to create ENIs with special settings.
+    type: dict
+    suboptions:
+      interfaces:
+        description:
+        - a list of ENI IDs (strings) or a list of objects containing the key I(id).
+        type: list
+      assign_public_ip:
+        description:
+        - when true assigns a public IP address to the interface
+        type: bool
+      private_ip_address:
+        description:
+        - an IPv4 address to assign to the interface
+        type: str
+      ipv6_addresses:
+        description:
+        - a list of IPv6 addresses to assign to the network interface
+        type: list
+      source_dest_check:
+        description:
+        - controls whether source/destination checking is enabled on the interface
+        type: bool
+      description:
+        description:
+        - a description for the network interface
+        type: str
+      private_ip_addresses:
+        description:
+        - a list of IPv4 addresses to assign to the network interface
+        type: list
+      subnet_id:
+        description:
+        - the subnet to connect the network interface to
+        type: str
   volumes:
     description:
     - A list of block device mappings, by default this will always use the AMI root device so the volumes option is primarily for adding more storage.
     - A mapping contains the (optional) keys device_name, virtual_name, ebs.volume_type, ebs.volume_size, ebs.kms_key_id,
       ebs.iops, and ebs.delete_on_termination.
     - For more information about each parameter, see U(https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_BlockDeviceMapping.html).
+    type: list
   launch_template:
     description:
       - The EC2 launch template to base instance configuration on.
-      - I(launch_template.id) the ID or the launch template (optional if name is specified).
-      - I(launch_template.name) the pretty name of the launch template (optional if id is specified).
-      - I(launch_template.version) the specific version of the launch template to use. If unspecified, the template default is chosen.
+    type: dict
+    suboptions:
+      id:
+        description:
+        - the ID or the launch template (optional if name is specified).
```

Comment:
I think this was a typo before
```suggestion
        - the ID of the launch template (optional if name is specified).
```